packages/@aws-cdk/aws-glue/lib/job.ts

```diff
@@ -0,0 +1,744 @@
+   *
+   * @default no security configuration.
+   */
+  readonly securityConfiguration?: ISecurityConfiguration;
+
+  /**
+   * The default arguments for this job, specified as name-value pairs.
+   *
+   * @see https://docs.aws.amazon.com/glue/latest/dg/aws-glue-programming-etl-glue-arguments.html for a list of  special parameters Used by AWS Glue
+   * @default no arguments
+   */
+  readonly defaultArguments?: { [key: string]: string };
+
+  /**
+   * The tags to use with this job.
+   *
+   * @default no tags
+   */
+  readonly tags?: { [key: string]: string };
+
+  /**
+   * The IAM role associated with this job.
+   *
+   * @default an IAM role is generated
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * Enables the collection of metrics for job profiling.
+   *
+   * @default - no profiling metrics emitted.
+   *
+   * @see `--enable-metrics` at https://docs.aws.amazon.com/glue/latest/dg/aws-glue-programming-etl-glue-arguments.html
+   */
+  readonly enableProfilingMetrics? :boolean;
+
+  /**
+   * Enables the Spark UI debugging and monitoring with the specified props.
+   *
+   * @default - Spark UI debugging and monitoring is disabled.
+   *
+   * @see https://docs.aws.amazon.com/glue/latest/dg/monitor-spark-ui-jobs.html
+   * @see https://docs.aws.amazon.com/glue/latest/dg/aws-glue-programming-etl-glue-arguments.html
+   */
+  readonly sparkUI?: SparkUIProps,
+
+  /**
+   * Enables Continuous Logging with the specified props.
+   *
+   * @default - Continuous Logging is disabled.
+   *
+   * @see https://docs.aws.amazon.com/glue/latest/dg/monitor-continuous-logging-enable.html
+   * @see https://docs.aws.amazon.com/glue/latest/dg/aws-glue-programming-etl-glue-arguments.html
+   */
+  readonly continuousLogging?: ContinuousLoggingProps,
+}
+
+/**
+ * A Glue Job.
+ */
+export class Job extends JobBase {
+  /**
+     * Creates a Glue Job
+     *
+     * @param scope The scope creating construct (usually `this`).
+     * @param id The construct's id.
+     * @param attrs Import attributes
+     */
+  public static fromJobAttributes(scope: constructs.Construct, id: string, attrs: JobAttributes): IJob {
+    class Import extends JobBase {
+      public readonly jobName = attrs.jobName;
+      public readonly jobArn = JobBase.buildJobArn(scope, attrs.jobName);
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * The ARN of the job.
+   */
+  public readonly jobArn: string;
+
+  /**
+   * The name of the job.
+   */
+  public readonly jobName: string;
+
+  /**
+   * The IAM role associated with this job.
+   */
+  public readonly role: iam.IRole;
+
+  /**
+   * The Spark monitoring configuration (Bucket, path) if Spark UI monitoring and debugging is enabled.
+   *
+   * @see https://docs.aws.amazon.com/glue/latest/dg/monitor-spark-ui-jobs.html
+   * @see https://docs.aws.amazon.com/glue/latest/dg/aws-glue-programming-etl-glue-arguments.html
+   */
+  public readonly sparkUIConfig?: SparkUIConfig;
```

Comment:
why are we saving this as a property in the first place?
```suggestion
  private readonly sparkUIConfig?: SparkUIConfig;
```

Reply:
To expose the created bucket - in case one was not explicitly provided

Reply:
Hmm...would it be nicer to somehow leverage the [existing CFN templates](https://docs.aws.amazon.com/glue/latest/dg/monitor-spark-ui-history.html) to create the UI instance here?

Reply:
So, the UI feature from a Spark job's point of view is all about producing UI logs in a bucket.
The page you refer 2 mentions 2 ways to run the server UI
1. Via a local docker container and pointing it at the bucket
2. Via the Cfn template which seems to spin up an EC2 instance with the UI server
Both methods require the bucket and the logs in it.